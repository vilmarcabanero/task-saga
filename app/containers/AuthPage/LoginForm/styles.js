import styled from 'styled-components';

export const Container = styled.div`
  .login-card {
    height: 27rem;
    max-width: 24rem;
    width: 100%;
    margin: auto;
    margin-top: 10rem;
    padding: 1.25rem;
    @media (max-width: 900px) {
      margin-top: 7rem;
    }

    @media (max-width: 767px) {
      margin-top: 2rem;
    }
  }

  .form {
    position: relative;
  }

  .error-container {
    display: flex;
    justify-content: center;
    align-items: center;
    position: absolute;
    bottom: -4.75rem;
    color: #d8000c;
    font-size: 18px;
    background-color: #ffbaba;
    border-radius: 3px;
    border: 1px solid rgba(216, 0, 12, 0.7);
    padding-left: 0.25rem;
    padding-right: 0.25rem;
    width: 100%;
  }

  .error {
    margin: 0.1rem;
    text-align: center;
    width: 80%;
  }

  .input-groups {
    margin-top: 1.5rem;
    position: relative;
    padding-bottom: 1rem;
  }

  .title {
    margin-top: 1.5rem;
    text-align: center;
  }

  .close-icon {
    position: absolute;
    right: 0.2rem;
    cursor: pointer;
    border-radius: 3px;
  }

  .close-icon:hover {
    background-color: rgba(255, 255, 255, 0.4);
  }

  .not-yet-registered {
    float: right;
  }
`;
