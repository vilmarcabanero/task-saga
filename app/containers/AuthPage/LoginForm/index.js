import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Close } from '@material-ui/icons';
import {
  Button as MUIButton,
  TextField,
  CircularProgress,
  Card,
} from '@material-ui/core';
import { Container } from './styles';
import { validateLogin } from '../validator';

export default function LoginForm(props) {
  const { setIsRegistered, login, getUser, userData, history } = props;
  const [loginUserData, setLoginUserData] = useState({
    email: '',
    password: '',
  });
  // const { loginUserData } = authPageState;
  const [error, setError] = useState('Invalid password.');
  const [isValid, setIsValid] = useState(true);
  const [doneLoggingIn, setDoneLoggingIn] = useState(true);

  const clearError = () => {
    setTimeout(() => {
      setIsValid(true);
    }, 4000);
  };

  const handleLogin = e => {
    e.preventDefault();
    const validateSuccess = validateLogin({
      loginUserData,
      setIsValid,
      setError,
    });

    if (validateSuccess) {
      login({
        loginUserData,
        setIsValid,
        setError,
        setDoneLoggingIn,
        history,
      });
    }

    clearError();
    // console.log(loginUserData);
    getUser();
    // console.log('userData', userData);
  };

  return (
    <Container>
      <Card className="login-card">
        <form onSubmit={handleLogin} className="form">
          {!userData.firstName ? (
            <h2 className="title">Login </h2>
          ) : (
            <h2 className="title"> {userData.firstName}</h2>
          )}
          <div className="input-groups pb-3 mt-3">
            <TextField
              type="text"
              label="Email address"
              fullWidth
              style={{ paddingBottom: '1rem' }}
              id="standard-basic-email"
              className="pb-3"
              value={loginUserData.email}
              onChange={e =>
                setLoginUserData({
                  ...loginUserData,
                  email: e.target.value,
                })
              }
            />

            <TextField
              type="password"
              label="Password"
              fullWidth
              id="standard-basic-password"
              style={{ paddingBottom: '1rem' }}
              value={loginUserData.password}
              onChange={e =>
                setLoginUserData({
                  ...loginUserData,
                  password: e.target.value,
                })
              }
            />

            <MUIButton style={{ marginBottom: '1rem' }}>
              Forgot password?
            </MUIButton>
            <MUIButton
              variant="contained"
              color="primary"
              type="submit"
              style={{ width: '100%', marginBottom: '1rem' }}
            >
              {doneLoggingIn ? (
                'Login'
              ) : (
                <CircularProgress size={24} style={{ color: '#fff' }} />
              )}
            </MUIButton>

            <MUIButton
              className="not-yet-registered"
              style={{ marginBottom: '1rem' }}
              onClick={() => setIsRegistered(false)}
            >
              Not yet registered?
            </MUIButton>
          </div>
          {!isValid && (
            <div className="error-container">
              <span className="error">{error}</span>
              <Close className="close-icon" onClick={() => setIsValid(true)} />
            </div>
          )}
        </form>
      </Card>
    </Container>
  );
}

LoginForm.propTypes = {
  history: PropTypes.func,
  getUser: PropTypes.func,
  userData: PropTypes.object,
  login: PropTypes.func,
  setIsRegistered: PropTypes.func,
};
