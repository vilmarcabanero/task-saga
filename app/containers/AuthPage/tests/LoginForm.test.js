/**
 * Test the HomePage
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import LoginForm from '../LoginForm';
import configureStore from '../../../configureStore';

describe('<LoginForm />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('Should render and match the snapshot', () => {
    const userData = {
      firstName: '',
    };
    const tree = renderer.create(
      <Provider store={store}>
        <IntlProvider locale="en">
          <LoginForm userData={userData} />
        </IntlProvider>
      </Provider>,
    );

    expect(tree).toMatchSnapshot();
  });
});
