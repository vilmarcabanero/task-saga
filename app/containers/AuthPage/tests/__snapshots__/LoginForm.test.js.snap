// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> Should render and match the snapshot 1`] = `
<div
  className="styles__Container-t4qiyr-0 hKNyRe"
>
  <div
    className="MuiPaper-root MuiCard-root login-card MuiPaper-elevation1 MuiPaper-rounded"
  >
    <form
      className="form"
      onSubmit={[Function]}
    >
      <h2
        className="title"
      >
        Login 
      </h2>
      <div
        className="input-groups pb-3 mt-3"
      >
        <div
          className="MuiFormControl-root MuiTextField-root pb-3 MuiFormControl-fullWidth"
          style={
            Object {
              "paddingBottom": "1rem",
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="standard-basic-email"
            id="standard-basic-email-label"
          >
            Email address
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="standard-basic-email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          style={
            Object {
              "paddingBottom": "1rem",
            }
          }
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink={false}
            htmlFor="standard-basic-password"
            id="standard-basic-password-label"
          >
            Password
          </label>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiInput-input"
              disabled={false}
              id="standard-basic-password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="password"
              value=""
            />
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "marginBottom": "1rem",
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Forgot password?
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "marginBottom": "1rem",
              "width": "100%",
            }
          }
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Login
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text not-yet-registered"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "marginBottom": "1rem",
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Not yet registered?
          </span>
        </button>
      </div>
    </form>
  </div>
</div>
`;
